File: 0001_0100/0027/0027.py
with open("input.txt", "r") as f, open("output.txt", "w") as q:
    w, h = map(int, f.readline().split())
    d = [1] * (w * h)
    f.readline()
    s = ([int(c) for c in x.split()] for x in f.read().split('\n')[:-1])
    for x in s:
        for i in range(x[1], x[3]):
            d[w * i + x[0] : w * i + x[2]] = [0] * (x[2] - x[0]) 
    q.write(str(sum(d)))

File: 0001_0100/0027/0027.go
package main

import (
	"fmt"
	"os"
)

func main() {
	f, _ := os.Open("input.txt")
	q, _ := os.Create("output.txt")
	var w, h, n, x1, y1, x2, y2 int
	fmt.Fscan(f, &w, &h, &n)
	d := make([]bool, w*h)
	for ; n > 0; n-- {
		fmt.Fscan(f, &x1, &y1, &x2, &y2)
		for ; x1 < x2; x1++ {
			for i := y1; i < y2; i++ {
				d[x1*w+i] = true
			}
		}
	}
	for i := range d {
		if d[i] {
			n++
		}
	}
	fmt.Fprint(q, w*h-n)
}
